From 066e15689d409d49140922d713c2a9983c58033b Mon Sep 17 00:00:00 2001
From: Roy Pledge <roy.pledge@nxp.com>
Date: Tue, 5 Sep 2017 16:47:49 -0400
Subject: [PATCH 164/741] drivers/staging/fsl_qbman: Fix SWP Mask in Error
 Handling

commit 541cb1d96c60d0d63a8c07cd56fe031b96189b02 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

If an ECC error occurs QBMan stores data in a register for
debug purposes. If the memory is software portal memory then
15 bits are used. This patch adjusts the mask to be correct.

Signed-off-by: Roy Pledge <roy.pledge@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/staging/fsl_qbman/qman_config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/fsl_qbman/qman_config.c b/drivers/staging/fsl_qbman/qman_config.c
index 9bb1e11a0a33..040631354df5 100644
--- a/drivers/staging/fsl_qbman/qman_config.c
+++ b/drivers/staging/fsl_qbman/qman_config.c
@@ -254,7 +254,7 @@ static const struct qman_error_info_mdata error_mdata[] = {
 	QMAN_ERR_MDATA(0x01FF, 72, "WQ context memory"),
 	QMAN_ERR_MDATA(0x00FF, 240, "CGR memory"),
 	QMAN_ERR_MDATA(0x00FF, 302, "Internal Order Restoration List memory"),
-	QMAN_ERR_MDATA(0x01FF, 256, "SW portal ring memory"),
+	QMAN_ERR_MDATA(0x7FFF, 256, "SW portal ring memory"),
 	QMAN_ERR_MDATA(0x07FF, 181, "CEETM class queue descriptor memory"),
 	QMAN_ERR_MDATA(0x0FFF, 140, "CEETM extended SFDR memory"),
 	QMAN_ERR_MDATA(0x0FFF, 25, "CEETM logical FQ mapping memory"),
-- 
2.17.1

